compiler/src/codegen.rs

```diff
@@ -40,19 +43,50 @@ impl Default for TagEnvironment {
 
 pub trait Backend: Default {
     fn tag_environment(&mut self, environment: TagEnvironment);
-    fn generate_type(&mut self, ty: &Type) -> Result<String>;
+    fn generate_type(
+        &mut self,
+        ty: &Type,
+        parent_prefix: Option<&Prefix>,
```

Comment:
I HATE this, but for prefixes of sequences, like `MainSequence ::= [APPLICATION 1] EXPLICIT SEQUENCE {...}`, the `APPLICATION 1` is parsed by the caller, but it is needed here, so it is passed down with each function call.